posydon/popsyn/binarypopulation.py

```diff
@@ -105,10 +107,20 @@ def __init__(self, **kwargs):
         entropy = self.kwargs.get('entropy', None)
         seq = np.random.SeedSequence(entropy=entropy)
 
-        self.comm = self.kwargs.pop('comm', None)
-        if self.comm is not None:
-            self.rank = self.comm.Get_rank()
-            self.size = self.comm.Get_size()
+        self.JOB_ID = self.kwargs.pop('JOB_ID', None)
+        if self.JOB_ID is not None:
+            self.rank = self.kwargs.pop('RANK', None)
+            self.size = self.kwargs.pop('size', None)
+            
+            # Make sure each of the MPI processes has the same entropy
+            # But unique per metallicity
+            if entropy is None:
+                met_shift = POSYDON_METALLICITIES.index(
+                                    self.kwargs.get('metallicity')) +1
+                
+                seq = np.random.SeedSequence(entropy=self.JOB_ID * met_shift)
```

Comment:
Usually I'd suggest to use a sum instead of a product to avoid getting too easy above the largest possible number.

Reply:
Would this provide enough uniqueness? ie. wouldn't we get similar seeds if the JOB_IDs of two runs are close together?

Reply:
This depends on the random number generator, which is used and how the entropy is exactly defined there. For most random number generators neighboring numbers are not that likely to be part of the same sequence. For a good random number generator, any connection (sum, product, any more complex functionality) between the seed should have the same likelihood to get two of them being part of the same sequence.

Reply:
I do not know if this really matters, since the randomness within a SyntheticPopulation is still fine.
But I was specifically referring to two different population synthesis runs.

For example, if JOB_1 has `JOB_ID =1` with 2 metallicities, then the entropy will be 1 and 2. 
If JOB_2 has 'JOB_ID=2', then the entropy will be 2 and 3. 
There will be overlap with the same entropy between those runs.

Reply:
You shouldn't get an overlap for the same metallicity. And overlaps for different metallicities shouldn't matter as long as you don't join them, which you shouldn't as long as you have fixed and dedicated metallicity values.
If you care about overlaps, I'd suggest you to use the JOB_ID of each individual job and not the one of the job array. Otherwise you can use the start time.
But all in all, it shouldn't matter too much how to shift it exactly.